subtrack.MAUI/Services/SubscriptionsCalculator.cs

```diff
@@ -95,4 +91,31 @@ public DateTime GetNextPaymentDate(Subscription subscription)
         var nextPaymentDate = GetNextPaymentDate(subscription);
         return (nextPaymentDate <= _dateProvider.Today, nextPaymentDate);
     }
+
+    private static bool PaidThisMonth(Subscription subscription, DateTime monthDate) => subscription.LastPayment.Month == monthDate.Month && subscription.LastPayment.Year == monthDate.Year;
+
+    private static bool PaidInPast(Subscription subscription, DateTime monthDate) => (subscription.LastPayment.Month < monthDate.Month && subscription.LastPayment.Year == monthDate.Year) || subscription.LastPayment.Year < monthDate.Year;
```

Comment:
weekly sub, last payment august 19 - next should be august 26 doesnt show up. Need to consider the occurrence of the sub. Probs need to change PaidThisMonth too